File: result/metagpt/claude/humaneval/11.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char* string_xor(const char* a, const char* b) {
    int len_a = strlen(a);
    int len_b = strlen(b);
    int max_len = len_a > len_b ? len_a : len_b;
    
    char* output = (char*)malloc(max_len + 1);
    if (output == NULL) {
        return NULL; // 内存分配失败
    }
    
    for (int i = 0; i < max_len; i++) {
        char char_a = i < len_a ? a[i] : '0';
        char char_b = i < len_b ? b[i] : '0';
        
        if (char_a == char_b) {
            output[i] = '0';
        } else {
            output[i] = '1';
        }
    }
    
    output[max_len] = '\0'; // 字符串结尾
    return output;
}

File: result/metagpt/gpt4o/humaneval/11.rs
fn string_xor(a: &str, b: &str) -> String {
    let len_a = a.len();
    let len_b = b.len();
    let max_len = len_a.max(len_b);

    let mut output = String::with_capacity(max_len);

    for i in 0..max_len {
        let char_a = if i < len_a { a.chars().nth(i).unwrap() } else { '0' };
        let char_b = if i < len_b { b.chars().nth(i).unwrap() } else { '0' };

        if char_a == char_b {
            output.push('0');
        } else {
            output.push('1');
        }
    }

    output
}

fn main() {
    let a = "1101";
    let b = "1011";
    let result = string_xor(a, b);
    println!("XOR result: {}", result);
}